---
output:
  md_document:
    variant: markdown_github
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, echo = FALSE, message=FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "README-"
)
library(ROKU)
```

# ROKU

An R implementation of ROKU, a method for identifying tissue-specific genes.

Details of the method are described in:

Kadota, K., Ye, J., Nakai, Y., Terada, T., Shimizu, K., 2006. [ROKU: a novel
method for identification of tissue-specific
genes](http://www.biomedcentral.com/1471-2105/7/294). BMC Bioinformatics 7, 294.

**Disclaimer**

This package was written as an personal exercise to practice implementing a 
published algorithm in R, and to learn now to create an R package. 
While some effort has been made to ensure accuracy of the algorithm, the package
has not been thoroughly tested and documented. Use at your own risk (see [LICENSE](LICENSE)).  

## Dependencies

ROKU uses the `affy` package, which can be installed from Bioconductor:

```{r eval=FALSE}
source("http://bioconductor.org/biocLite.R")
biocLite("affy")
```

## Installation

```r
install.packages("devtools")
devtools::install_github("kcha/ROKU")
```

## Usage

ROKU requires as input a data frame of expression values where each row is a gene
and each column is a tissue/sample. For example: 

```{r}
head(psidata)
```

To find tissue-specific genes:
```{r}
rk <- ROKU(psidata)
df <- ROKU_as_df(rk)
head(df)
```

Genes with a non-zero values are considered outliers.
